Reserve vector in StringCordView ctor

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151682
* #151697
* #151630
* #151629
* __->__ #151628
* #151627
* #151626

Clear missing reserve (we should expect that pieces are not empty).

Differential Revision: [D73129445](https://our.internmc.facebook.com/intern/diff/D73129445/)

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

swolchok committed:
Reserve vector in StringCordView ctor

Clear missing reserve (we should expect that pieces are not empty).

Differential Revision: [D73129445](https://our.internmc.facebook.com/intern/diff/D73129445/)

[ghstack-poisoned]

malfet commented on torch/csrc/jit/frontend/source_range.cpp:
```diff
@@ -17,6 +17,7 @@ StringCordView::StringCordView(
     std::vector<std::string_view> inputs,
     std::vector<std::shared_ptr<std::string>> ownerships)
     : pieces_(std::move(inputs)), owned_strings_(std::move(ownerships)) {
+  accumulated_sizes_.reserve(pieces_.size() + 1);
```
Please add a comment why this is needed/better

swolchok commented:
> we should expect that pieces are not empty

by the way, I think either StringCordView itself or StringCordView + Lexer is buggy when some pieces are empty, and we shouldn't be attempting to skip empty pieces. hopefully I will get around to looking into that later in this stack.

swolchok committed:
Update on "Reserve vector in StringCordView ctor"


Clear missing reserve (we should expect that pieces are not empty).

Differential Revision: [D73129445](https://our.internmc.facebook.com/intern/diff/D73129445/)

cc EikanWang jgong5 wenzhe-nrv sanchitintel

[ghstack-poisoned]
